cpp/include/cudf/copying.hpp

```diff
@@ -79,5 +79,78 @@ std::unique_ptr<column> allocate_like(column_view input, size_type size,
  */
 std::unique_ptr<table> empty_like(table_view input_table);
 
+/**
+ * @brief Slices a column_view (including null values) into a set of column_views
+ * according to a set of indices.
+ *
+ * The `slice` function creates `column_view`s from `input` with multiple intervals
```

Comment:
Here's a start on cleaning up and simplifying this documentation. 

```suggestion
 * The returned view's of `input` are constructed from an even number indices where
 * the `i`th returned `column_view` will view the elements in `input` indicated by the range
 * `[indices[i], indices[i+1])`.
 * 
 * For all `i` it is expected `indices[i] <= input.size()`
 * For all `i%2==0`, it is expected that `indices[i] <= indices[i+1]`
```